redis/src/types.rs

```diff
@@ -1372,20 +1428,36 @@ macro_rules! from_redis_value_for_tuple {
                 // hacky way to count the tuple size
                 let mut n = 0;
                 $(let $name = (); n += 1;)*
-                if items.len() % n != 0 {
-                    invalid_type_error!(items, "Bulk response of wrong dimension")
-                }
-
-                // this is pretty ugly too.  The { i += 1; i - 1} is rust's
-                // postfix increment :)
                 let mut rv = vec![];
                 if items.len() == 0 {
                     return Ok(rv)
                 }
-                for chunk in items.chunks_exact(n) {
```

Comment:
can you please explain in words the changes you've made here and above? I don't understand them.

Reply:
as an example test_zunionstore_weights will fail without these lines, because old code expects
`(bulk(string-data('"one"'), string-data('"5"'), string-data('"three"'), string-data('"9"'), string-data('"two"'), string-data('"10"'))` 
resp3 gives 
`bulk(bulk(string-data('"one"'), double(5.0)), bulk(string-data('"three"'), double(9.0)), bulk(string-data('"two"'), double(10.0))))`
so even items.len() % n clause will fail here.
btw i'm not good at writing macros

Reply:
Well, I'm not good at reading macros, so we'll just have to learn :)

I see, so you're flattening the list and then chunking it? It seems like you have an assumption that the list either contains only `Bulk` values, or no `Bulk` values. If a list is mixed, you'll only return the flattened bulking.
If this assumption is what you want to work with, then please assert it, so that it would be documented clearly.

Reply:
It's Vector of tuples so i don't expect to get Set or Map